lib/astarte_vmq_plugin/rpc/server.ex

```diff
@@ -70,12 +70,30 @@ defmodule Astarte.VMQ.Plugin.RPC.Server do
   @impl GenServer
   def handle_call({:delete, %{realm_name: realm, device_id: device}}, _from, state) do
     client_id = "#{realm}/#{device}"
-    # Either the client has been deleted or it is :not_found,
-    # which means that there is no session anyway.
-    Plugin.disconnect_client(client_id, true)
-    Plugin.ack_device_deletion(realm, device)
 
-    {:reply, :ok, state}
+    # Step 1: Disconnect client (best effort - ignore errors since client may not be connected)
+    case Plugin.disconnect_client(client_id, true) do
+      :ok ->
+        :ok
+
+      # Client wasn't connected, that's fine
+      {:error, :not_found} ->
+        :ok
+    end
```

Comment:
Since we're `:ok`ing it anyway, we can just remove the `case`
```suggestion
    Plugin.disconnect_client(client_id, true)
```